src/api/providers/cline.ts

```diff
@@ -1,143 +1,233 @@
-				throw new Error(`Cline API Error ${error.code}: ${error.message}${metadataStr}`)
-			}
+		const url = `${this._baseUrl}/api/v1/chat/completions`
+		try {
+			const response = await axios.post(
+				url,
+				{
+					model: this.getModel().id,
+					messages: [
+						{
+							role: "system",
+							content: systemPrompt,
+						},
+						...messages,
+					],
+					stream: false,
+					// reasoning_effort: this.options.reasoningEffort || "low",
+					// thinking_budget_tokens: this.options.thinkingBudgetTokens || 0,
+					// open_router_provider_sorting: this.options.openRouterProviderSorting || "default",
+				},
+				requestConfig,
+			)
 
-			if (!this.lastGenerationId && chunk.id) {
-				this.lastGenerationId = chunk.id
+			if (!response.data || !response.data.data) {
+				throw new Error(`Request to ${url} failed with status ${response.status}`)
 			}
 
-			// Check for mid-stream error via finish_reason
-			const choice = chunk.choices?.[0]
-			// OpenRouter may return finish_reason = "error" with error details
-			if ((choice?.finish_reason as string) === "error") {
-				const choiceWithError = choice as any
-				if (choiceWithError.error) {
-					const error = choiceWithError.error
-					console.error(`Cline Mid-Stream Error: ${error.code || error.type || "Unknown"} - ${error.message}`)
-					throw new Error(`Cline Mid-Stream Error: ${error.code || error.type || "Unknown"} - ${error.message}`)
-				} else {
-					throw new Error("Cline Mid-Stream Error: Stream terminated with error status but no error details provided")
-				}
+			if (!response.data.data.choices || response.data.data.choices.length === 0) {
+				throw new Error(`No choices returned from Cline API: ${JSON.stringify(response.data)}`)
 			}
 
-			const delta = choice?.delta
-			if (delta?.content) {
-				yield {
-					type: "text",
-					text: delta.content,
+			for (const choice of response.data.data.choices) {
+				if (choice.finish_reason === "error") {
+					const error = choice.error || { code: "Unknown", message: "No error details provided" }
+					console.error(`Cline API Error: ${error.code} - ${error.message}`)
+					throw new Error(`Cline API Error: ${error.code} - ${error.message}`)
 				}
-			}
-
-			// Reasoning tokens are returned separately from the content
-			if ("reasoning" in delta && delta.reasoning) {
-				yield {
-					type: "reasoning",
-					// @ts-ignore-next-line
-					reasoning: delta.reasoning,
+				if (choice.delta && choice.delta.content) {
+					yield {
+						type: "text",
+						text: choice.delta.content,
+					}
 				}
-			}
-
-			if (!didOutputUsage && chunk.usage) {
-				// @ts-ignore-next-line
-				let totalCost = (chunk.usage.cost || 0) + (chunk.usage.cost_details?.upstream_inference_cost || 0)
-				const modelId = this.getModel().id
-				const provider = modelId.split("/")[0]
-
-				// If provider is x-ai, set totalCost to 0 (we're doing a promo)
-				if (provider === "x-ai") {
```

Comment:
Are your changes post the xAI promo? Do we need to mitigate the xAI specific changes to work with the new API? @pashpashpash has been working with the xAI team so we'll need to make sure that our updated API works with the expectations on their server about requests from us (ie in the header and body)

Reply:
Thanks for the reminder. @pashpashpash and I had touched base on this. I'll work on that tonight

Reply:
Got a list of things to handle here.